src/coreclr/src/gc/gc.cpp

```diff
@@ -18088,12 +18088,14 @@ uint8_t* gc_heap::find_object (uint8_t* interior)
     {
         // this is a pointer to a UOH object
         heap_segment* seg = find_segment (interior, FALSE);
-        if (seg
+        if (seg)
+        {
 #ifdef FEATURE_CONSERVATIVE_GC
-            && (GCConfig::GetConservativeGC() || interior <= heap_segment_allocated(seg))
+            if (interior >= heap_segment_allocated(seg))
+                return 0;
+#else
+            assert (interior < heap_segment_allocated (seg));
```

Comment:
```suggestion
```
The exact same assert is repeated unconditionally a few lines below. I would delete the #el

Reply:
Thanks, removed.